api/src/main/java/org/openmrs/module/fhir2/api/translators/CodingTranslator.java

```diff
@@ -11,11 +11,19 @@
 
 import javax.annotation.Nonnull;
 
+import lombok.Setter;
+import org.hl7.fhir.r4.model.CodeableConcept;
 import org.hl7.fhir.r4.model.Coding;
 import org.hl7.fhir.r4.model.ICoding;
 import org.openmrs.Concept;
+import org.openmrs.util.OpenmrsUtil;
+import org.springframework.beans.factory.annotation.Autowired;
 
-public interface CodingTranslator extends ToFhirTranslator<Concept, Coding>, ToOpenmrsTranslator<Concept, ICoding> {
+@Setter
+public abstract class CodingTranslator implements ToFhirTranslator<Concept, Coding>, ToOpenmrsTranslator<Concept, ICoding> {
```

Comment:
Do we need to turn this into an abstract class? I'd rather leave it as an interface and move the coding here into the actual implementation class.

Reply:
The usual pattern we use for this would be to add a `BaseCodingTranslator` class that we could extend for the actual implementations. (The idea of the interfaces is to keep things as agnostic of the underlying code as possible; this enables implementations to swap out our implementations for... w/e they want—otherwise, we wouldn't have all these interfaces!)

Reply:
so i went ahead and switched to a BaseCodingTranslator abstract class and CodingTranslator interface since that's the standard we've been using. let me know if that's what you had in mind.